### DISCUSSION TITLE
(DONE) Need help reviewing SSL patches

### DISCUSSION THREAD

**Discussion by brandon-rhodes** [XEphem/XEphem#7] [General]
[**Update:** The patches are now landed and XEphem now supports SSL!]

Hey, everyone! There are two user-contributed patches to XEphem that I would like to commit to `main` but that need independent review — if you have more recent practice than I do reading C code to look for buffer overflows, please jump in with your opinion at this issue:

https://github.com/XEphem/XEphem/issues/5

Thanks!

**Comment by obspsr** [comment]
Hi Brandon,
I discovered your message after today's mine !

I applied the 2 patches of Ludwig on my fresh fork it worked perfectly as in precedent 3.7.7:

...  in XEphem/
patch -p3 < xephem-3.7.7_openssl_earthmenu.patch
patch -p3 < xephem-3.7.7_openssl.patch 
I just had to add -lssl at GUI/xephem/Makefile line 37 ...

my patched actual version is from your 4.0.2 as you can check from
https://github.com/obspsr/XEphem.git

Good night !
Serge.

**Reply by brandon-rhodes** [reply]
I'm glad that the patches worked for you! They worked for me as well. Do you know anyone who could attempt a security audit of the new code? It would be nice to have some assurance that the code is safe before we merge it.

**Reply by obspsr** [reply]
I can assure you that nobody called about ssl problems fore more than 2 years ! and actually if you git clone here, we have no Sun
images no database downloads as all addresses are wrong. so it is urgent to merge this code definitively... I think ...
May be we could as Elwood to test/validate my fork on a MAC as I think he use it ...?
Best regards. Serge.

**Reply by brandon-rhodes** [reply]
The SSL patches are now merged and with the recent 4.1.0 release, Sun images should now download successfully. Once again, thank you, @lmaendle for providing the new code that supports SSL!
